src/librustc/traits/object_safety.rs

```diff
@@ -67,9 +67,9 @@ impl ObjectSafetyViolation {
     }
 
     pub fn span(&self) -> Option<Span> {
-        match self {
+        match *self {
             ObjectSafetyViolation::AssocConst(_, span) |
-            ObjectSafetyViolation::Method(_, _, span) => Some(*span),
+            ObjectSafetyViolation::Method(_, _, span) if span != DUMMY_SP => Some(span),
```

Comment:
Can you explain this change? Why does `DUMMY_SP` occur?

Reply:
When the trait is from a different crate. One of the tests that got `bless`ed in the last commit recover a `note` that had gone away. Sadly `DUMMY_SP` are way too common still.

Reply:
Ah; could you transcribe your note here to a comment in the code?